lcdb/reporting.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+""" Reporting System """
+
+import os
+import sys
+import base64
+from textwrap import dedent
+
+import markdown
+import jinja2
+
+TEMPLATES_DIR = os.path.join(os.path.dirname(__file__), 'templates')
+loader = jinja2.FileSystemLoader(TEMPLATES_DIR)
+ENV = jinja2.Environment(loader=loader)
+
+def report(string, **kwargs):
+    """ Build a report """
+    md = markdown.Markdown(output_format='html5')
+    tp = ENV.get_template('base.html')
+    return tp.render(content=md.convert(dedent(string).format(**kwargs)))
+
+
+class Report(object):
+    """ Basic report class """
+    pass
+
+
+class Section(object):
+    """ Basic section class """
+    pass
+
+
+class Image(object):
+    """ Image class """
+
+    def __init__(self, png, name='', caption=''):
+        with open(png, 'rb') as fh:
+            uri = base64.b64encode(fh.read()).decode('utf-8').replace('\n', '')
+            self.image = 'data:image/png;base64,{uri}'.format(uri=uri)
+            self.image_html = '<a href={0}><img src="{0}"/></a>'.format(self.image)
+
```

Comment:
Seems reasonable and useful. If size becomes an issue we can move `image` and `image_html` to properties (`@property` decorator) so that the URI is in memory only once rather than 3x here.